spec/models/range_request_spec.rb

```diff
@@ -116,23 +137,105 @@
     expect(info[:is_multipart]).to be_truthy
 
     expect(info[:range_start_bytes][0]).to eq(0)
-    expect(info[:range_end_bytes][0]).to eq(range_request.max_range_size)
+    expect(info[:range_end_bytes][0]).to eq(range_request.max_range_size - 1)
 
     expect(info[:range_start_bytes][1]).to eq(0)
-    expect(info[:range_end_bytes][1]).to eq(range_request.max_range_size)
+    expect(info[:range_end_bytes][1]).to eq(range_request.max_range_size - 1)
 
     # last 654321 bytes (or last max_range_size, which ever is smaller)
-    expect(info[:range_start_bytes][2]).to eq(audio_file_mono_size_bytes - 500 - 1)
+    expect(info[:range_start_bytes][2]).to eq(audio_file_mono_size_bytes - 500)
     expect(info[:range_end_bytes][2]).to eq(audio_file_mono_size_bytes - 1)
 
     expect(info[:range_start_bytes][3]).to eq(0)
     expect(info[:range_end_bytes][3]).to eq(10)
 
     expect(info[:range_start_bytes][4]).to eq(50)
-    expect(info[:range_end_bytes][4]).to eq(range_request.max_range_size + 50)
+    expect(info[:range_end_bytes][4]).to eq(range_request.max_range_size + 50 - 1)
+  end
 
-    expect(info[:range_start_bytes][1]).to eq(0)
-    expect(info[:range_end_bytes][1]).to eq(range_request.max_range_size)
+  context 'special open end range case' do
+    # this test case comes from a real-world production bug: https://github.com/QutBioacoustics/baw-server/issues/318
+    # the second part of a large range request triggers a negative content length and negative gradient in the content
+    # range header.
+
+
+    # before bug fix:
+    # file_size:                                  822281
+    # request:                      "Range: bytes 512001-"
+    # info[:range_start]:                         512001
+    # info[:range_end]:                           310279       <-- problem, negative range!
```

Comment:
well, to be clear, the issue with the `:range_end` is that it is less than the `:range_start`